### DISCUSSION TITLE
socketio 随便使用一个正则都会报错

### DISCUSSION THREAD

**Discussion by ReallyCreep** [midwayjs/midway#3760] [Q&A] [answered]
### Describe the problem（描述问题）

import {
  WSController,
  App,
  OnWSConnection,
  OnWSDisConnection,
  Inject,
  OnWSMessage,
  WSEmit,
} from '@midwayjs/decorator';
import { Context, Application } from '@midwayjs/socketio';

/**
 * socket
 */
@WSController(/^\/[0-9a-fA-F]$/)
export class socketController {
  @Inject()
  ctx: Context;

  @App('socketIO')
  app: Application;

  /**
   * 客户端连接
   */
  @OnWSConnection()
  async onConnection() {
    console.log('on client connect', this.ctx.id);
    console.log('参数', this.ctx.handshake.query);
    this.ctx.emit('data', '连接成功1s23');
    const a = this.app.of('');
    a.adapter.on('error', err => {
      console.log('ee');
    });
  }
  /**
   * 客户端断开连接
   */
  @OnWSDisConnection()
  async onDisConnection() {
    console.log('on client connect', this.ctx.id);
  }
  // 消息事件
  @OnWSMessage('myEvent')
  @WSEmit('myEventResult')
  async gotMessage(data) {
    console.log('on data got', this.ctx.id, data);
    return data + '__abc';
  }
}
异常错误为：

start error: nsp.adapter.on is not a function
TypeError: nsp.adapter.on is not a function
    at MidwaySocketIOFramework.addNamespace *\node_modules\.pnpm\@midwayjs+socketio@3.15.2\node_modules\@midwayjs\socketio\dist\framework.js:188:25)       

可以修改   framework.js 中的

`  if (nsp.adapter) {
            nsp.adapter.on('error', err => {
                this.logger.error(err);
            });
        }`
修改为`  if (nsp.adapter) {
            nsp.on('error', err => {
                this.logger.error(err);
            });
        }`就正常了，是什么原因导致的呢？

### Midway Versions（Midway 版本）

Start to check your midway component version...

✓ @midwayjs/faas-typings(not installed)
✓ @midwayjs/fc-starter(not installed)
✓ @midwayjs/serverless-http-parser(not installed)     
✓ @midwayjs/async-hooks-context-manager(not installed)
✓ @midwayjs/axios(not installed)
✓ @midwayjs/bootstrap(3.15.0)
✓ @midwayjs/bull(not installed)
✓ @midwayjs/bull-board(not installed)
✓ @midwayjs/cache-manager(3.15.2)
✓ @midwayjs/captcha(not installed)
✓ @midwayjs/casbin(not installed)
✓ @midwayjs/casbin-redis-adapter(not installed)
✓ @midwayjs/casbin-typeorm-adapter(not installed)
✓ @midwayjs/code-dye(not installed)
✓ @midwayjs/consul(not installed)
✓ @midwayjs/core(3.15.0)
✓ @midwayjs/cos(not installed)
✓ @midwayjs/cron(3.15.2)
✓ @midwayjs/cross-domain(3.15.2)
✓ @midwayjs/decorator(3.15.0)
✓ @midwayjs/etcd(not installed)
✓ @midwayjs/express-session(not installed)
✓ @midwayjs/faas(not installed)
✓ @midwayjs/grpc(not installed)
✓ @midwayjs/http-proxy(not installed)
✓ @midwayjs/i18n(not installed)
✓ @midwayjs/info(3.15.2)
✓ @midwayjs/jwt(not installed)
✓ @midwayjs/kafka(not installed)
✓ @midwayjs/leoric(not installed)
✓ @midwayjs/mikro(not installed)
✓ @midwayjs/mock(3.15.2)
✓ @midwayjs/mongoose(not installed)
✓ @midwayjs/mqtt(not installed)
✓ @midwayjs/oss(not installed)
✓ @midwayjs/otel(not installed)
✓ @midwayjs/passport(not installed)
✓ @midwayjs/process-agent(not installed)
✓ @midwayjs/prometheus(not installed)
✓ @midwayjs/prometheus-socket-io(not installed)
✓ @midwayjs/rabbitmq(not installed)
✓ @midwayjs/redis(not installed)
✓ @midwayjs/security(not installed)
✓ @midwayjs/sequelize(not installed)
✓ @midwayjs/session(not installed)
✓ @midwayjs/socketio(3.15.2)
✓ @midwayjs/static-file(3.15.2)
✓ @midwayjs/swagger(3.15.2)
✓ @midwayjs/tablestore(not installed)
✓ @midwayjs/tags(not installed)
✓ @midwayjs/typegoose(not installed)
✓ @midwayjs/typeorm(3.15.2)
✓ @midwayjs/upload(3.15.2)
✓ @midwayjs/validate(3.15.2)
✓ @midwayjs/version(not installed)
✓ @midwayjs/view(not installed)
✓ @midwayjs/view-ejs(3.15.2)
✓ @midwayjs/view-nunjucks(not installed)
✓ @midwayjs/web(not installed)
✓ @midwayjs/express(not installed)
✓ @midwayjs/koa(3.15.2)
✓ @midwayjs/ws(not installed)
**********************************************************************
>> Check complete, all versions are healthy.
**********************************************************************

**Comment by czy88840616** [comment, marked as answer]
https://github.com/midwayjs/midway/pull/3764

**Comment by ReallyCreep** [comment]
解决的非常及时。希望能尽快发布版本。

**Reply by czy88840616** [reply]
https://github.com/midwayjs/midway/releases/tag/v3.15.8
